aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/CreateHandler.java

```diff
@@ -0,0 +1,163 @@
+package com.amazonaws.iot.accountauditconfiguration;
+
+import java.util.Map;
+import java.util.Set;
+
+import software.amazon.awssdk.services.iot.IotClient;
+import software.amazon.awssdk.services.iot.model.AuditNotificationTarget;
+import software.amazon.awssdk.services.iot.model.DescribeAccountAuditConfigurationRequest;
+import software.amazon.awssdk.services.iot.model.DescribeAccountAuditConfigurationResponse;
+import software.amazon.awssdk.services.iot.model.IotException;
+import software.amazon.awssdk.services.iot.model.UpdateAccountAuditConfigurationRequest;
+import software.amazon.awssdk.utils.CollectionUtils;
+import software.amazon.awssdk.utils.StringUtils;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+public class CreateHandler extends BaseHandler<CallbackContext> {
+
+    private final IotClient iotClient;
+
+    public CreateHandler() {
+        iotClient = IotClient.builder().build();
+    }
+
+    @Override
+    public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            AmazonWebServicesClientProxy proxy,
+            ResourceHandlerRequest<ResourceModel> request,
+            CallbackContext callbackContext,
+            Logger logger) {
+
+        ResourceModel model = request.getDesiredResourceState();
+
+        // We ask customers to specify the Account ID as part of the model,
+        // because there must be a primary identifier.
+        // Having it in the model helps CFN prevent cases like 2 stacks from managing the same resource.
+        String accountIdFromTemplate = model.getAccountId();
+        String accountId = request.getAwsAccountId();
+        if (!accountIdFromTemplate.equals(accountId)) {
+            return ProgressEvent.failed(model, callbackContext, HandlerErrorCode.InvalidRequest,
+                    String.format("AccountId in the template (%s) doesn't match actual: %s.",
+                            accountIdFromTemplate, accountId));
+        }
+
+        // Call Describe to see whether the customer has a configuration already.
+        // Note that this API never throws ResourceNotFoundException.
+        // If the configuration doesn't exist, it returns an object with all checks disabled,
+        // other fields equal to null.
+        DescribeAccountAuditConfigurationResponse describeResponse;
+        try {
+            describeResponse = proxy.injectCredentialsAndInvokeV2(
+                    DescribeAccountAuditConfigurationRequest.builder().build(),
+                    iotClient::describeAccountAuditConfiguration);
+        } catch (IotException e) {
+            throw Translator.translateIotExceptionToCfn(e);
```

Comment:
Will these exceptions be logged somewhere for debugging purposes

Reply:
Yes, the framework logs the exception and the stack trace.